test/integration/compound/compound_test.go

```diff
@@ -0,0 +1,426 @@
+
+	"github.com/gardener/controller-manager-library/pkg/ctxutil"
+	"github.com/gardener/external-dns-management/pkg/apis/dns/v1alpha1"
+	"github.com/gardener/external-dns-management/pkg/controller/provider/mock"
+)
+
+var _ = Describe("Compound controller tests", func() {
+	const (
+		defaultTTL   = 300
+		retryTimeout = 5 * time.Second
+	)
+
+	var (
+		testRunID      string
+		testNamespace  *corev1.Namespace
+		firstZoneID    dns.ZoneID
+		e1, e2, e3, e4 *v1alpha1.DNSEntry
+
+		checkDeleted = func(g Gomega, ctx context.Context, entry *v1alpha1.DNSEntry) {
+			err := testClient.Get(ctx, client.ObjectKeyFromObject(entry), entry)
+			g.Expect(err).To(HaveOccurred())
+			g.Expect(client.IgnoreNotFound(err)).To(Succeed())
+		}
+
+		checkSingleEntryInMockDatabaseGomega = func(g Gomega, entry *v1alpha1.DNSEntry) {
+			dump := mock.TestMock[testRunID].BuildFullDump()
+			for _, zoneDump := range dump.InMemory {
+				switch {
+				case zoneDump.HostedZone.Domain == "first.example.com" && entry == nil:
+					g.Expect(zoneDump.DNSSets).To(BeEmpty(), "unexpected number of DNS sets in first.example.com")
+				case zoneDump.HostedZone.Domain == "first.example.com" && entry != nil:
+					g.Expect(zoneDump.DNSSets).To(HaveKey(dns.DNSSetName{DNSName: entry.Spec.DNSName}))
+					g.Expect(zoneDump.DNSSets).To(HaveLen(1), "unexpected number of DNS sets in first.example.com")
+					set := zoneDump.DNSSets[dns.DNSSetName{DNSName: entry.Spec.DNSName}]
+					g.Expect(set.Sets).To(HaveKey("A"))
+					g.Expect(set.Sets).To(HaveKey("META"))
+					g.Expect(set.Sets).To(HaveLen(2))
+					setA := set.Sets["A"]
+					g.Expect(setA.Records).To(ConsistOf(
+						PointTo(MatchFields(IgnoreExtras, Fields{
+							"Value": Equal(entry.Spec.Targets[0]),
+						}))))
+					g.Expect(setA.Type).To(Equal("A"))
+					g.Expect(setA.TTL).To(Equal(int64(defaultTTL)))
+				case zoneDump.HostedZone.Domain == "second.example.com":
+					g.Expect(zoneDump.DNSSets).To(BeEmpty(), "unexpected number of DNS sets in second.example.com")
+				default:
+					g.Expect(false).To(BeTrue(), "unexpected zone domain "+zoneDump.HostedZone.Domain)
+				}
+			}
+		}
+
+		checkSingleEntryInMockDatabase = func(entry *v1alpha1.DNSEntry) {
+			checkSingleEntryInMockDatabaseGomega(Default, entry)
+		}
+
+		checkEntry = func(entry *v1alpha1.DNSEntry) {
+			Eventually(func(g Gomega) {
+				Expect(testClient.Get(ctx, client.ObjectKeyFromObject(entry), entry)).To(Succeed())
+				g.Expect(entry.Status.State).To(Equal("Ready"))
+				if entry.Spec.Targets != nil {
+					g.Expect(entry.Status.Targets).To(Equal(entry.Spec.Targets))
+				} else {
+					g.Expect(entry.Status.Targets).To(Equal(quoted(entry.Spec.Text)))
+				}
+				if entry.Spec.TTL != nil {
+					g.Expect(entry.Status.TTL).To(Equal(entry.Spec.TTL))
+				} else {
+					g.Expect(entry.Status.TTL).To(Equal(ptr.To[int64](defaultTTL)))
+				}
+				g.Expect(entry.Status.ObservedGeneration).To(Equal(entry.Generation))
+			}).Should(Succeed())
+		}
```

Comment:
Same as above + L85 should also be using the scoped Gomega instance, right?
```suggestion
		checkEntry = func(entry *v1alpha1.DNSEntry) {
			Eventually(func(g Gomega) {
				g.ExpectWithOffset(1, testClient.Get(ctx, client.ObjectKeyFromObject(entry), entry)).To(Succeed())
				g.ExpectWithOffset(1, entry.Status.State).To(Equal("Ready"))
				if entry.Spec.Targets != nil {
					g.ExpectWithOffset(1, entry.Status.Targets).To(Equal(entry.Spec.Targets))
				} else {
					g.ExpectWithOffset(1, entry.Status.Targets).To(Equal(quoted(entry.Spec.Text)))
				}
				if entry.Spec.TTL != nil {
					g.ExpectWithOffset(1, entry.Status.TTL).To(Equal(entry.Spec.TTL))
				} else {
					g.ExpectWithOffset(1, entry.Status.TTL).To(Equal(ptr.To[int64](defaultTTL)))
				}
				g.ExpectWithOffset(1, entry.Status.ObservedGeneration).To(Equal(entry.Generation))
			}).Should(Succeed())
		}
```